src/prefect/serialization/flow.py

```diff
@@ -16,21 +16,43 @@
 )
 
 
-def get_parameters(obj: prefect.Flow, context: dict) -> Set:
+def get_parameters(obj: prefect.Flow, context: dict) -> List:
     if isinstance(obj, prefect.Flow):
-        return {p for p in obj.tasks if isinstance(p, prefect.core.parameter.Parameter)}
+        params = {
+            p for p in obj.tasks if isinstance(p, prefect.core.parameter.Parameter)
+        }
     else:
-        return utils.get_value(obj, "parameters")
+        params = utils.get_value(obj, "parameters")
 
+    return sorted(params, key=lambda p: p.slug)
 
-def get_reference_tasks(obj: prefect.Flow, context: dict) -> Set:
+
+def get_reference_tasks(obj: prefect.Flow, context: dict) -> List:
     if isinstance(obj, prefect.Flow):
-        return obj._reference_tasks
+        tasks = obj._reference_tasks
     else:
-        return utils.get_value(obj, "reference_tasks")
+        tasks = utils.get_value(obj, "reference_tasks")
+
+    return sorted(tasks, key=lambda t: t.slug)
+
+
+def get_tasks(obj: prefect.Flow, context: dict) -> List:
+    return list(sorted(obj.tasks, key=lambda t: t.slug))
+
+
+def get_edges(obj: prefect.Flow, context: dict) -> List:
+    return list(
+        sorted(
+            obj.edges,
+            key=lambda e: f"{e.upstream_task.slug}-{e.downstream_task.slug}",
```

Comment:
No need to format this, tuples are sortable just fine.

```suggestion
            key=lambda e: (e.upstream_task.slug, e.downstream_task.slug),
```